### ISSUE TITLE
Create mock/hardcoded "custom actions" for testing skill implementation

### ISSUE THREAD

**Issue by PaulTreitel** [uchicago-cs/chiventure#557]
This should break down into roughly 3 phases:

1. Read the documentation from the custom actions team and discuss with them as necessary so that we fully understand the kinds of behaviors that custom actions will support.
2. Produce some code that mimics some of that behavior through hardcoding (eg implement a "change stat" action by directly changing the stat's value).
3. Assemble these hardcoded behaviors into sequences that could realistically represent the kinds of things done by a single skill.

This should be thorough enough that the code and the hardcoded examples can be used to test the functions of an actual skill module.

**Comment by PaulTreitel** [comment]
I've created a branch for this, [playerclass/custom-action-skills](https://github.com/uchicago-cs/chiventure/tree/playerclass/custom-action-skills), and written some [basic code](https://github.com/uchicago-cs/chiventure/blob/playerclass/custom-action-skills/src/playerclass/sandbox/custom_skills.c) for it. It contains mockup structs for representing simple custom actions and can assemble those structs to represent two mock skills, swing_sword, and fireball_spell. There is an extremely rudimentary interpreter that can execute these skills, but's it's extremely limited in functionality. It can only handle a few commands and works by modifying a single stats struct (stubbing for a real one).

**Comment by PaulTreitel** [comment]
I've refactored the code in the branch into a few files so it makes more sense. It now has a custom_skills.h file with the struct definitions, a custom_skills.c file with the basic creator functions and an execute function. The hardcoded skills are now in custom_skills_main.c. I think the code is roughly ready for review, so I've started a PR as well.

**Comment by PaulTreitel** [comment]
I've implemented data structures representing very simple custom actions (if/else/endif and add/subtract). A highly simplified execution function allows these custom actions to be run. The if command provides the ability to check if a stat is equal to a certain value. The if, else, and endif commands together allow for branching custom action execution (but not nested branches). The add and subtract commands allow a stat to be changed by a constant amount. These data structures are assembled into simple hardcoded skills which are then executed. This proof-of-concept code is now merged into dev.
